universal-application-tool-0.0.1/app/controllers/admin/AdminProgramController.java

```diff
@@ -45,7 +47,11 @@ public AdminProgramController(
 
   @Secure(authorizers = Authorizers.Labels.UAT_ADMIN)
   public Result index() {
-    return ok(listView.render(this.service.listProgramDefinitions()));
+    return ok(
+        listView.render(
+            this.service.listProgramDefinitions().stream()
+                .sorted(Comparator.comparing(program -> program.lifecycleStage()))
+                .collect(ImmutableList.toImmutableList())));
```

Comment:
Suggestion (blocking): The sort order of the program definitions is a presentation concern, recommend moving this into the view. Could also see adding it to the service. Either way, not a concern of the controller, which should be focused on handling HTTP and brokering between views and service classes.

Reply:
More on that here: https://github.com/seattle-uat/civiform/wiki/Dev-guide-&-standards#separation-of-concerns